lib/dap/input.rb

```diff
@@ -50,17 +51,23 @@ def read_record
   # JSON Input (line-delimited records)
   #
   class InputJSON
-    
+
     include FileSource
 
     def initialize(args)
       self.open(args.first)
     end
 
-    def read_record  
+    def read_record
       line = self.fd.readline rescue nil
       return Error::EOF unless line
-      json = Oj.load(line.strip) rescue nil
+      begin
+        json = Oj.load(line.strip)
+      rescue
+        $stderr.puts "\nRecord is not valid JSON and will be skipped."
```

Comment:
Why the beginning of line newline and then putting the faulty line on its own?  It might make for a better post-processing experience for users of dap if you just did it all on one line: `$stderr.puts "Skipping invalid JSON: #{line}"`.  `puts` will already put the newline for you.